src/istio/mixerclient/report_batch.cc

```diff
@@ -78,7 +78,8 @@ void ReportBatch::FlushWithLock() {
   // pointer so at least the memory will be freed if this lambda is deleted
   // without being called, but really this should be a unique_ptr that is
   // moved into the transport_ and then moved into the lambda if invoked.
-  transport_(request, &*response, [this, response](const Status& status) {
+  auto shared_this = shared_from_this();
+  transport_(request, &*response, [this, shared_this, response](const Status& status) {
```

Comment:
Just a question, why capture both this and shared_this?

Reply:
We need to capture shared_this because it prevents "this" from being deleted before the closure/std::function runs.  We capture "this" because it makes the resulting closure/std::function run logically within a member function context which means we can access private variables and functions.